lib/rules/no-restricted-imports.js

```diff
@@ -218,8 +236,11 @@ module.exports = {
             groupedRestrictedPaths[importSource].forEach(restrictedPathEntry => {
                 const customMessage = restrictedPathEntry.message;
                 const restrictedImportNames = restrictedPathEntry.importNames;
+                const restrictedImportNamePattern = restrictedPathEntry.importNamePattern ? new RegExp(restrictedPathEntry.importNamePattern, "u") : null;
+                const allowedImportNames = restrictedPathEntry.allowImportNames;
+                const allowedImportNamePattern = restrictedPathEntry.allowImportNamePattern ? new RegExp(restrictedPathEntry.allowImportNamePattern, "u") : null;
```

Comment:
`paths` objects don't have `restrictedImportNamePattern` and  `allowedImportNamePattern` options.

Reply:
@mdjermanovic These should be resolved now, tested and committed.